### ISSUE TITLE
New DB wait script prevents container from starting

### ISSUE THREAD

**Issue by N-Schaef** [TandoorRecipes/recipes#1656]
### Tandoor Version

1.1.4

### Setup

Docker / Docker-Compose

### Reverse Proxy

Traefik

### TL;DR: 
Somehow in the container for me the default postgres user variable is broken and prevents my container from starting.
Adding: `PGUSER=<databaseuser>`to the .env file fixes my startup issues.

### Long version:

Since updating the docker image my container does not start anymore.
I've seen similar issues in this repo, but most seem to have been a network/config misconfiguration.
I can't guarantee that this case is different, but I could not find any mistakes.

The container stops with the following log messages:
```
Checking configuration...
Waiting for database to be ready...

Database not reachable. Maximum attempts exceeded.
Please check logs above - misconfiguration is very likely.
Make sure the DB container is up and POSTGRES_HOST is set properly.
Shutting down container.
```

My PostgreSQL (11) database is running outside of any docker container as a normal debian installation.
It is configured to only accept connections from specific users from specific IPs (relatively standard configuration).

When entering the recipes container and trying to connect to the DB with:
```bash
psql --host=${POSTGRES_HOST} --port=${POSTGRES_PORT} --user=${POSTGRES_USER} -W
```
I can connect to the DB just fine.
But when using the DB wait script of the container:
```bash
pg_isready --host=${POSTGRES_HOST} --port=${POSTGRES_PORT}
```
I get the following output
```bash
<Static-IP-To-Server>:5432 - no attempt
```

While digging through the [source code](https://github.com/postgres/postgres/blob/master/src/bin/scripts/pg_isready.c) of pg_isready and searching through the [relevant](https://docs.huihoo.com/doxygen/postgresql/libpq-fe_8h.html#af0fce1f5992d2c7c7b0c3aa28f304644) [functions](https://docs.huihoo.com/doxygen/postgresql/fe-connect_8c.html#a81cdb2f1cd25adf0db94b9088bf88122) to end up with this output I noticed that this only happens if the connection parameters can't be parsed.

pg_isready uses default parameters, which are overwritten by the supplied parameters.
So I've tried overwriting random parameters, and noticed that if I supply a `--user` parameter (any content, does not have to be a valid user), it suddenly works.
So I checked what went wrong:
The function for fetching default parameters can be found [here](https://docs.huihoo.com/doxygen/postgresql/fe-connect_8c.html#a511085dcf1b8f4bac16606115dca26b7).
It first tries to fetch them from a PostgreSQL service spec (I did not find it in any of the container).
And then from the environment variables specified [here](https://docs.huihoo.com/doxygen/postgresql/fe-connect_8c_source.html) (for the user variable `PGUSER`).
But this environment variable is empty in the container.

The empty user variable is invalid and results in the "no attempt" message.
For example
```bash
pg_isready --host=${POSTGRES_HOST} --port=${POSTGRES_PORT} --user=""
```
also results in "no attempt"
but 
```bash
pg_isready --host=${POSTGRES_HOST} --port=${POSTGRES_PORT} --user="randomstring"
```
does not.

To be honest, I don't know why it only crashes for me but not for other people, which is why I don't rule out a configuration error.
But I've managed to get my container running by supplying `PGUSER=<myDBUser>` to the .env file.
Another possibility would be to supply the (anyway required) `POSTGRES_USER` attribute to the `pg_isready` command in the wait script.

If you don't want to modify this script then it's also ok, I've just wanted to post my investigation if others run into the same problem.

**Comment by MaxJa4** [comment]
Adding it doesn't hurt the "regular" configurations with the dockerized psql instance, so why not imo.

**Comment by duracell** [comment]
Hey, it's not crashing only for you and great job figuring out what's the problem.
I just tried to install TandoorRecipes for myself and ran into the same problem.
Hope this will get fixed soon, because the default setup may be good for most people, but with many containers a database container for everything is not the best solution in terms of performance, backup and simplicity. :)

**Comment by Zadeis** [comment]
I am also having this problem with my database being on a separate system and can't connect as of version 1.1.0 and any subsequent version. I am currently stuck on version 1.0.8 as it starts without issue. 

My data is on a separate system and functions fine on 1.0.8. When I turn on debug info no further information is provided by the logs. 

I have not changed anything about my configuration upgrading between versions. If I can provide further information let me know how to and I'll try my best to assist you with diagnosing this problem.

**Comment by N-Schaef** [comment]
Well, to check if you have the same problem as this issue is about, you can add `PGUSER=<your database user>` to the .env file or pass it as an environment variable to your docker-compose/docker deployment.
If it works, you probably have the same problem, which may be fixed.

**Comment by Zadeis** [comment]
That seems to have solved the database problem however now I am getting a 502 bad gateway so I think something else is borked now. Unless you can think of as to why given the circumstance I'll continue diagnosing further since this new problems seems out of scope of the original issue. 

Much thanks for your hard work on this software!

**Comment by vabene1111** [comment]
can you post the logs of your container when getting the 502 and also when starting up without the PGUSER variable so i can see whats going on there
